### PR TITLE
Add extension support

### PR DESCRIPTION
**Pull request by dignifiedquire**
- Browser tests
- Extension support (needed for https://github.com/ipfs/js-ipfs-ipld)

Depends on:
- [x] #4 
- [x] https://github.com/ipfs/js-ipfs-repo/pull/36

cc @diasdavid

### CODE REVIEW

**Comment by daviddias** [comment]
Can we make the 'extension' be transparent and have a type instead, as for 'ipld', and 'protobuf', from UX perspective, it will be easier to read that we want to read the block that is stored as protobuf or as ipld, rather than having to know the extension (which .data doesn't really mean anything).

File: README.md [hunk 1]
```diff
@@ -122,29 +131,29 @@ bs.getBlock(multihash, function (err, block) {
...
     // block successfuly deleted
   }
 })
 ```
 
-##### deleteBlocks
+##### `deleteBlocks`
 
-```JavaScript
+```js
```
Comment (daviddias) [inline README.md:156]: master nicpicker :P

**Comment by dignifiedquire** [comment]
Not sure I agree with that. If we go this route we have to encode the mapping of type to extension into the `Block` and need to change it everytime we might want to add a new type. If we just use extension we can pass any new thing through without ever having to touch the internals of `Block` again.

**Comment by daviddias** [comment]
I understand, hopefully, we won't to change again though :). It does make an `js-ipfs` contributor have to have domain knowledge of how blocks are put in disk.

File: src/block.js [hunk 2]
```diff
@@ -18,7 +18,20 @@ function Block (data, extension) {
...
+    switch (this.type) {
+      case 'protobuf':
+        return 'data'
+      case 'ipld':
+        return 'ipld'
+      default:
+        return this.type
+    }
+  }
+})
```
Comment (daviddias) [inline src/block.js:35]: nice way to do it :)

**Pull request merged**